flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -172,6 +181,88 @@ public AbstractYarnClusterDescriptor(
 		userJarInclusion = getUserJarInclusionMode(flinkConfiguration);
 
 		this.configurationDirectory = Preconditions.checkNotNull(configurationDirectory);
+
+		String yarnQueueConfigValue = flinkConfiguration.getString(YarnConfigOptions.YARN_QUEUE);
```

Comment:
The difficulty of this ticket (FLINK-8799) is that `flinkConfiguration` is mutable. As long as a reference of `flinkConfiguration` can possibly leak, this class remains mutable. Also, there are some private methods that mutate the configuration. There are several places where we would need to make defensive copies, e.g., in the constructor.